### PR TITLE
Switch to the `AssociatedLegendrePolynomials.jl` package

### PR DESCRIPTION
**Pull request by jiahe23**
This PR 
- [x] switches use the `AssociatedLegendrePolynomials.jl` package for current spherical harmonics computation;
- [x] adds the vector tests of diffusion and hyperdiffusion to `runtests.jl`;
- [x] cleans up the hyperdiffusion scalar test.

### CODE REVIEW

File: test/sphere_sphericalharmonics.jl [hunk 1]
```diff
@@ -1,22 +1,8 @@
...
 # input lat and long in degrees
 
 function Ylm(l, m, lat, long)
-    return real(computeYlm(deg2rad(90 - lat), deg2rad(long), lmax = l)[(l, m)])
-end
-
-function Plm(l, m, x)
-    return computePlmx(x, lmax = l)[(l, m)] *
-           sqrt(2 * π * factorial(l + m) / (2 * l + 1) / factorial(l - m))
+    return real(λlm(l, m, sind(lat)) * exp(im * m * deg2rad(long)))
```
Comment (simonbyrne) [inline test/sphere_sphericalharmonics.jl:5]: This should be equivalent to
```suggestion
    return λlm(l, m, sind(lat)) * cosd(m * long)
```

**Comment by jiahe23** [comment]
bors try

**Comment by jiahe23** [comment]
bors try

**Review by simonbyrne — APPROVED** [review]

**Comment by simonbyrne** [comment]
@jakebolewski any idea what is going on here?

**Comment by jakebolewski** [comment]
it looks like the xvfb process can hang after the command terminates causing this issue :-/

let me try a workaround

**Review by jakebolewski — APPROVED** [review]

**Pull request merged**